crates/block-producer/src/store/mod.rs

```diff
@@ -44,9 +44,11 @@ pub struct TransactionInputs {
     ///
     /// We use NonZeroU32 as the wire format uses 0 to encode none.
     pub nullifiers: BTreeMap<Nullifier, Option<NonZeroU32>>,
-    /// List of unauthenticated notes that were not found in the store
-    pub missing_unauthenticated_notes: Vec<NoteId>,
-    /// The current block height
+    /// Unauthenticated notes which are present in the store.
+    ///
+    /// These are notes which were committed _after_ the transaction was created.
+    pub found_unauthenticated_notes: BTreeSet<NoteId>,
```

Comment:
This is fine for now, but in the future, we may also need to get the authentication paths for these notes.